lib/rack/query_parser.rb

```diff
@@ -85,11 +85,29 @@ def parse_nested_query(qs, separator = nil)
     # the structural types represented by two different parameter names are in
     # conflict, a ParameterTypeError is raised.
     def normalize_params(params, name, v, depth)
-      raise RangeError if depth <= 0
+      raise RangeError if depth >= param_depth_limit
+
+      if !name
```

Comment:
Can youn please add documentation explaining the behaviour of the implementation, even if just high level, e.g.

```
# Parse and consume the X token:
```

Reply:
Basically, if anyone else needs to work on this code, let's make sure they are with relative ease, able to understand the high level structure and how it's working.

Reply:
That makes sense, I pushed a commit that adds some comments.